aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -802,7 +801,7 @@ module aptos_token::token {
     }
 
     public fun create_royalty(royalty_points_numerator: u64, royalty_points_denominator: u64, payee_address: address): Royalty {
-        assert!(royalty_points_denominator > 0, error::invalid_argument(EINVALID_ROYALTY_NUMERATOR_DENOMINATOR));
+        assert!(royalty_points_denominator >= 0, error::invalid_argument(EINVALID_ROYALTY_NUMERATOR_DENOMINATOR));
```

Comment:
while more permissive, I think this is very dangerous... to divide by 0 is dangerous and makes this assert completely meaningless... honestly, the assert should probably just be `!= 0` so as not to trick you into believing negative numbers exist.

Reply:
For the 1st comment, Any reason to believe the royalty_denominator smaller than numerator is not a malicious  use-case? currently, only 639 out of 1,187,257 token have royalty_denominator smaller than numerator. I think the impact would positive and is only limited to 0.005% potentially malicious use-cases. I don't know if we should go through AIP for this case. 

For the 2nd comment, we don't enforce royalty_points_denominator to be non 0 during token creation now. There are a non-trivial percentage of token has 0 denominator (mostly ANS token). I found the community already has a work around to treat denominator being 0 as no royalty.  This _create_royalty_  function is only used in mutation flow. The change in this PR is to unify the creation and mutation using one create_royalty function.  And then enforce the denominator to be non zero through an AIP.

Reply:
I see.... then we should just delete this all together